actors/builtin/multisig/multisig_test.go

```diff
@@ -75,10 +78,179 @@ func TestConstruction(t *testing.T) {
 	})
 }
 
-func newAddr(t *testing.T, id uint64) addr.Address {
+func TestPropose(t *testing.T) {
+	actor := multisig.MultiSigActor{}
+
+	receiver := newIDAddr(t, 100)
+	anne := newSecpAddr(t, 101)
+	bob := newSecpAddr(t, 102)
+	chuck := newSecpAddr(t, 103)
+	richard := newSecpAddr(t, 105)
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple propose", func(t *testing.T) {
+		rt := builder.Build()
+		constructParams := multisig.ConstructorParams{
+			Signers:               []addr.Address{anne, bob},
+			NumApprovalsThreshold: 2,
+			UnlockDuration:        0,
+		}
+
+		rt.ExpectValidateCallerAddr(builtin.InitActorAddr)
+		constructRet := actor.Constructor(rt, &constructParams)
+		assert.Equal(t, runtime.EmptyReturn{}, *constructRet)
+		rt.Verify()
+
+		// going to assume the state is correct, a method that constructs and verifies would be nice.
+		proposeParams := &multisig.ProposeParams{
+			To:     chuck,
+			Value:  abi.NewTokenAmount(10),
+			Method: 0,
+			Params: nil,
+		}
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		proposeRet := actor.Propose(rt, proposeParams)
+		assert.Equal(t, multisig.ProposeReturn{multisig.TxnID(0)}, *proposeRet)
+		rt.Verify()
+
+		// validate the actor state and transaction exist
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		assert.Equal(t, constructParams.Signers, st.Signers)
+		assert.Equal(t, constructParams.NumApprovalsThreshold, st.NumApprovalsThreshold)
+		assert.Equal(t, abi.NewTokenAmount(0), st.InitialBalance)
+		assert.Equal(t, abi.ChainEpoch(0), st.UnlockDuration)
+		assert.Equal(t, abi.ChainEpoch(0), st.StartEpoch)
+
+		txns := adt.AsMap(rt.Store(), st.PendingTxns)
+		keys, err := txns.CollectKeys()
+		require.NoError(t, err)
+		// there is exactly one transaction
+		assert.Equal(t, 1, len(keys))
+
+		var txn multisig.MultiSigTransaction
+		found, err := txns.Get(keys[0], &txn)
+		assert.True(t, found)
+
+		// containing these values
+		assert.Equal(t, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(10),
+			Method:   0,
+			Params:   abi.MethodParams{},
+			Approved: []addr.Address{anne},
+		}, txn)
+	})
+
+	t.Run("propose with threshold met", func(t *testing.T) {
+		// TODO implement mockrt message send
+		t.SkipNow()
```

Comment:
the mock runtime does not implement `Send()`, yet. Skip this test until its implemented.